src/test/java/mate/academy/security/AuthenticationServiceImplTest.java

```diff
@@ -0,0 +1,68 @@
+package mate.academy.security;
+
+import static org.mockito.ArgumentMatchers.any;
+import static org.mockito.ArgumentMatchers.anyString;
+
+import java.util.Optional;
+import java.util.Set;
+import mate.academy.exception.AuthenticationException;
+import mate.academy.model.Role;
+import mate.academy.model.User;
+import mate.academy.service.RoleService;
+import mate.academy.service.UserService;
+import org.junit.jupiter.api.Assertions;
+import org.junit.jupiter.api.BeforeEach;
+import org.junit.jupiter.api.Test;
+import org.mockito.Mockito;
+import org.springframework.security.crypto.password.PasswordEncoder;
+
+class AuthenticationServiceImplTest {
+    private static final String TEST_EMAIL_OK = "artem@gmail.com";
+    private static final String TEST_PASSWORD_OK = "12345678";
+    private User user;
+    private Role role;
+    private UserService userService;
+    private RoleService roleService;
+    private PasswordEncoder passwordEncoder;
+    private AuthenticationService authenticationService;
+
+    @BeforeEach
+    void setUp() {
+        user = new User();
+        role = new Role(Role.RoleName.USER);
+        user.setEmail(TEST_EMAIL_OK);
+        user.setPassword(TEST_PASSWORD_OK);
+        user.setRoles(Set.of(role));
+        userService = Mockito.mock(UserService.class);
+        roleService = Mockito.mock(RoleService.class);
+        passwordEncoder = Mockito.mock(PasswordEncoder.class);
+        authenticationService = new AuthenticationServiceImpl(userService, roleService,
+                passwordEncoder);
+    }
```

Comment:
Please do not reinstantiate objects/mock before each tests unless necessary for state reset (like for dao)
```suggestion
class AuthenticationServiceImplTest {
    private static final String TEST_EMAIL_OK = "artem@gmail.com";
    private static final String TEST_PASSWORD_OK = "12345678";
    private User user;
    private Role role;
    private UserService userService = mock(UserService.class);
    private RoleService roleService = mock(RoleService.class);
    private PasswordEncoder passwordEncoder = mock(PasswordEncoder.class);
    private AuthenticationService authenticationService = 
            new AuthenticationServiceImpl(userService, roleService, passwordEncoder);

    @BeforeEach
    void setUp() {
        user = new User();
        role = new Role(Role.RoleName.USER);
        user.setEmail(TEST_EMAIL_OK);
        user.setPassword(TEST_PASSWORD_OK);
        user.setRoles(Set.of(role));
    }
```